e2e/tests/test_21_api_metrics.py

```diff
@@ -20,7 +20,14 @@ def test_metrics() -> None:
     assert response.status_code == 200, f"{response.status_code} - {response.text}"
     content = response.text
     lines = content.split("\n")
-    metrics = {line.split(" ")[0]: float(line.split(" ")[1]) for line in lines if line and line[0] != "#"}
+    # examples:
+    # starlette_requests_total{method="GET",path_template="/metrics"} 1.0
+    # method_steps_processing_time_seconds_sum{method="healthcheck_endpoint",step="all"} 1.6772013623267412e-05
+    metrics = {
+        parts[0]: float(parts[1])
+        for line in lines
+        if line and line[0] != "#" and len(parts := line.rsplit(" ", 1)) == 2
```

Comment:
Nice refactoring, using the assignment expression. This was introduced in Python 3.8, but as we work with Python 3.9.15, it is safe to use it.

One question, why do you replace `split` with `rsplit`? Do you expect that `parts[0]` might contain any blanks?

Maybe we can make the refactoring simpler with something like this?
```suggestion
        if line and line[0] != "#" and (parts := line.rsplit(" ", 1))
```

Reply:
> One question, why do you replace split with rsplit? Do you expect that parts[0] might contain any blanks?

Yes, exactly, we have spaces in the keys